prov/rxd/src/rxd_cq.c

```diff
@@ -978,9 +1006,12 @@ static void rxd_handle_op(struct rxd_ep *ep, struct rxd_pkt_entry *pkt_entry)
 	if (ep->peers[base_hdr->peer].peer_addr == FI_ADDR_UNSPEC)
 		goto release;
 
-	rx_entry = rxd_unpack_init_rx(ep, pkt_entry, base_hdr, &sar_hdr,
-				      &tag_hdr, &data_hdr, &rma_hdr, &atom_hdr,
-				      &msg, &msg_size);
+	ret = rxd_unpack_init_rx(ep, &rx_entry, pkt_entry, base_hdr, &sar_hdr,
+				 &tag_hdr, &data_hdr, &rma_hdr, &atom_hdr,
+				 &msg, &msg_size);
+	if (ret)
+		goto ack;
+
 	if (!rx_entry) {
```

Comment:
Can rx_entry ever be null here now?

Reply:
Yes, if it's null and a msg/tagged op, that means it went to the unexpected queue, and if it's an rma/atomic op, there wasn't enough memory to allocate a new entry so it does not process the op (see below)